kale/pipeline/multi_domain_adapter.py

```diff
@@ -0,0 +1,313 @@
+# =============================================================================
```

Comment:
Some docstrings at the top will be helpful to give an overview. See other examples, except deep_dti.